tachyon/math/polynomials/univariate/radix2_evaluation_domain.h

```diff
@@ -440,6 +447,31 @@ class Radix2EvaluationDomain
     }
   }
 
+  // Expands a |Eigen::MatrixBase|'s rows from |rows| to |rows|^(|added_bits|),
+  // moving values from row |i| to row |i|^(|added_bits|). All new entries are
+  // set to |F::Zero()|. Returns false when |added_bits| is zero, and no
+  // expansion is performed.
+  template <typename Derived, typename Derived2>
```

Comment:
It appears as though mat and out are both a `RowMajorMatrix<F>` and therefore the same type! `Derived2` seems like it can be replaced by `Derived`.

Reply:
There needs to be some steps to make them same type. In the commit that this change was made, the types were different. If you look at `two_adic_fri_impl.h`, `CosetLDEBatch()` used `Eigen::Map<tachyon::math::RowMajorMatrix<F>>` for `mat` and `tachyon::math::RowMajorMatrix<F>` for `out`. They became same after the commit `feat!: update c API of `CosetLDEBatch()` to accept pre-allocated memory`, so I'll make them same in this commit.